src/devices/logitech_g930.c

```diff
@@ -18,19 +31,57 @@ void g930_init(struct device** device)
 
     strncpy(device_g930.device_name, "Logitech G930", sizeof(device_g930.device_name));
 
-    device_g930.capabilities = CAP_SIDETONE;
+    device_g930.capabilities = CAP_SIDETONE | CAP_BATTERY_STATUS;
     device_g930.send_sidetone = &g930_send_sidetone;
+    device_g930.request_battery = &g930_request_battery;
 
     *device = &device_g930;
 }
 
 static int g930_send_sidetone(hid_device* device_handle, uint8_t num)
 {
-#define MSG_SIZE 64
     uint8_t data[MSG_SIZE] = { 0xFF, 0x0A, 0, 0xFF, 0xF4, 0x10, 0x05, 0xDA, 0x8F, 0xF2, 0x01, num, 0, 0, 0, 0 };
 
     for (int i = 16; i < MSG_SIZE; i++)
         data[i] = 0;
 
     return hid_send_feature_report(device_handle, data, MSG_SIZE);
 }
+
+static int g930_request_battery(hid_device* device_handle)
+{
+    /*
+        CREDIT GOES TO https://github.com/Roliga for the project
+        https://github.com/Roliga/g930-battery-percentage/blob/master/main.c
+        I've simply ported that implementation to this project!
+    */
+
+    int i;
+    int res;
+
+    unsigned char buf[MSG_SIZE] = { 0xFF, 0x09, 0x00, 0xFD, 0xF4, 0x10, 0x05, 0xB1, 0xBF, 0xA0, 0x04 };
+    for (i = 11; i < MSG_SIZE; i++)
+        buf[i] = 0;
+
+    res = hid_send_feature_report(device_handle, buf, MSG_SIZE);
+    if (res < 0) {
+        fprintf(stderr, "Failed to send feature report: %ls\n", hid_error(device_handle));
+        return -1;
+    }
+
+    for (i = 0; i < 3; i++) {
+        res = hid_get_feature_report(device_handle, buf, MSG_SIZE);
+
+        if (res < 0) {
+            fprintf(stderr, "Failed to get feature report %d: %ls\n", i, hid_error(device_handle));
+            return -1;
```

Comment:
```suggestion
            return res;
```

The main function does the error handling and will also print the error.